auslib/db.py

```diff
@@ -319,6 +319,10 @@ def insert(self, changed_by=None, transaction=None, **columns):
 
            @rtype: sqlalchemy.engine.base.ResultProxy
         """
+        if dryrun:
```

Comment:
Okay, that unrelated then :) 

I was thinking of making the early return of dry-run happen just before the actual transaction happens. Something like:

```
def insert(self, changed_by=None, transaction=None, dryrun=False, **columns):
    if self.history and not changed_by: 
         raise ValueError("changed_by must be passed for Tables that have history") 

    if dryrun:
         self.log.debug("In dryrun mode, no transaction executed...")
         return

      if transaction:
            return self._prepareInsert(transaction, changed_by, **columns)
      else:
           with AUSTransaction(self.getEngine()) as trans:
              return self._prepareInsert(trans, changed_by, **columns)
```

Would it provide some value?